python/pyspark/sql/tests/test_functions.py

```diff
@@ -294,6 +294,12 @@ def test_input_file_name_reset_for_rdd(self):
         for result in results:
             self.assertEqual(result[0], '')
 
+    def test_input_file_name_udf(self):
+        df = self.spark.read.text('python/test_support/hello/hello.txt')
+        df = df.select(udf(lambda x: x)("value"), input_file_name().alias('file'))
+        file_name = df.collect()[0].file
+        self.assertTrue("python/test_support/hello/hello.txt" in file_name)
```

Comment:
Can't we add a test case in the scala side by using `IntegratedUDFTestUtils`? (I checked the other physical plans using threads (e.g., `SubqueryExec` and `BroadcastExchangeExec`) though, I couldn't find a test query to reproduce this..)

Reply:
We can add but I think the test will be virtually same as the current test. This one test should be fine.